### PR TITLE
update referring to changeRequests from pull14038 containing:

### PR DESCRIPTION
**Pull request by renebaudisch**
render ads that are available but not rendered when the slot is above the fold and then scrolled into view
https://github.com/ampproject/amphtml/pull/14038/commits/8bbef696fa1e5dab078f12e1ee98a089b1b2ad14

render ads above the fold when scrolled to view
https://github.com/ampproject/amphtml/pull/14038/commits/197b1aa12559b819cd2c122749e35e46d1bdafb4

handle nobid, use promise-like pattern
https://github.com/ampproject/amphtml/pull/14038/commits/c336f63d51a6f7d153e855196d2190418d57605e

fix lint errors
https://github.com/ampproject/amphtml/pull/14038/commits/29d29d73333a8730d7197e6b04e36d05687943fb

remove whitespace
https://github.com/ampproject/amphtml/pull/14038/commits/65e2304c034c0b8dfd99af653302f717baceafcb

doc apntag.showTag
https://github.com/ampproject/amphtml/pull/14038/commits/a153ed593fef3323b1ab93733cd52aa6b4c11426

de-lint
https://github.com/ampproject/amphtml/pull/14038/commits/f88083e113074ffdf4bd3955f07094c32281fac9

fix check-types
https://github.com/ampproject/amphtml/pull/14038/commits/a5f31e3cb1a8818ba4847cfe03ddd5493a2cbdd9

🐛 Bug fix (`:bug:`)

### CODE REVIEW

**Comment by renebaudisch** [comment]
Please assign to bradfrizzell and set liamqma, mkendall07 and matthewlane as reviewers

**Comment by bradfrizzell** [comment]
@liamqma @matthewlane

File: ads/appnexus.js [hunk 1]
```diff
@@ -24,103 +24,111 @@ const APPNEXUS_AST_URL = 'https://acdn.adnxs.com/ast/ast.js';
...
-   * encodeURIComponent);
-   * @param  {!Array<string>} args query string params to add to the base URL.
-   * @return {string}      Formated TTJ URL.
-   */
-  function constructTtj(args) {
-    let url = 'https://ib.adnxs.com/ttj?';
-    for (let i = 0; i < args.length; i++) {
-      //append arg to query. Please encode arg first.
-      url += args[i] + '&';
+    const args = [];
```
Comment (bradfrizzell) [inline ads/appnexus.js]: your lint is wrong throughout this file, see Travis failures https://travis-ci.org/ampproject/amphtml/jobs/404814579

Please fix (which will also cause many of these lines to have no diff)

File: ads/appnexus.js [hunk 2]
```diff
@@ -24,103 +24,111 @@ const APPNEXUS_AST_URL = 'https://acdn.adnxs.com/ast/ast.js';
...
-  appnexusAst(global, data);
+    appnexusAst(global, data);
 
 }
 
-/**
- * @param {!Window} global
- * @param {!Object} data
- */
 function appnexusAst(global, data) {
```
Comment (bradfrizzell) [inline ads/appnexus.js]: add back removed jsdoc.

Comment (renebaudisch) [reply ads/appnexus.js]: oh, sure, just copied from 14038, will add it agan

**Comment by renebaudisch** [comment]
comprehensibly, at least the review is done now and we do have a mergable PR now

**Comment by mkendall07** [comment]
hey @renebaudisch thanks grabbing those commits for us! This looks good, I'm just going to test it quickly today and reply shortly.

**Comment by mkendall07** [comment]
Test and LGTM.
@bradfrizzell 
can be merged.

**Comment by bradfrizzell** [comment]
Apologies for the delay, I was OOO yesterday

**Comment by renebaudisch** [comment]
That's no problem at all, I'm waiting for a few month now, so a single day doesn't hurt anymore :)

**Comment by liamqma** [comment]
Hi,

Thanks All! 

I am not familiar with AMP deployment process. Once this PR is merged, will this change appear in AMP production bundle: https://cdn.ampproject.org/v0/amp-ad-0.1.js?

**Comment by renebaudisch** [comment]
Still not working :(

**Comment by renebaudisch** [comment]
When you open the page, please wait a second until the page has finished loading. Then scroll down, a fixed bottomAd Space will appear and within that update it now should render but it does not, therefore now it throws an error and complains that apntag is not defined anymore?
image

But there is a check on apntag a few lines before...!?!

image

**Comment by mkendall07** [comment]
Hi @renebaudisch 
Thanks for the details. The problem with the test page is you cannot currently single and multi ad units on the same page. 

Example: 
Single ad unit
```
<amp-ad width="300" height="250" type="appnexus" data-member="7823" data-code="welt.de-amp-ros-mrec" data-target="mrec_2"> `
with
```
Muilti ad unit placement:
```
<amp-ad width="320" height="50"
--
  | type="appnexus"
  | data-target="mrec_sticky"
  | json="{&quot;pageOpts&quot;:{&quot;member&quot;:7823},&quot;adUnits&quot;:[{&quot;disablePsa&quot;:true,&quot;invCode&quot;:&quot;welt.de-amp-ros-mrec&quot;,&quot;sizes&quot;:[[300,250]],&quot;keywords&quot;:{&quot;kw_misc&quot;:[],&quot;pt0&quot;:96449,&quot;pt1&quot;:&quot;werbemitteltest2&quot;},&quot;targetId&quot;:&quot;mrec&quot;},{&quot;disablePsa&quot;:true,&quot;invCode&quot;:&quot;welt.de-amp-ros-mrec&quot;,&quot;sizes&quot;:[[]],&quot;keywords&quot;:{&quot;kw_misc&quot;:[],&quot;pt0&quot;:96449,&quot;pt1&quot;:&quot;werbemitteltest2&quot;},&quot;targetId&quot;:&quot;teads&quot;},{&quot;disablePsa&quot;:true,&quot;invCode&quot;:&quot;welt.de-amp-ros-mrec&quot;,&quot;sizes&quot;:[[300,250]],&quot;keywords&quot;:{&quot;kw_misc&quot;:[],&quot;pt0&quot;:96449,&quot;pt1&quot;:&quot;werbemitteltest2&quot;},&quot;targetId&quot;:&quot;mrec_sticky&quot;},{&quot;disablePsa&quot;:true,&quot;invCode&quot;:&quot;welt.de-amp-ros-mrec&quot;,&quot;sizes&quot;:[[300,250]],&quot;keywords&quot;:{&quot;kw_misc&quot;:[],&quot;pt0&quot;:96449,&quot;pt1&quot;:&quot;werbemitteltest2&quot;},&quot;targetId&quot;:&quot;mrec_2&quot;},{&quot;disablePsa&quot;:true,&quot;invCode&quot;:&quot;welt.de-amp-ros-mrec&quot;,&quot;sizes&quot;:[[300,250]],&quot;keywords&quot;:{&quot;kw_misc&quot;:[],&quot;pt0&quot;:96449,&quot;pt1&quot;:&quot;werbemitteltest2&quot;},&quot;targetId&quot;:&quot;mrec_3&quot;},{&quot;disablePsa&quot;:true,&quot;invCode&quot;:&quot;welt.de-amp-ros-mrec&quot;,&quot;sizes&quot;:[[300,250]],&quot;keywords&quot;:{&quot;kw_misc&quot;:[],&quot;pt0&quot;:96449,&quot;pt1&quot;:&quot;werbemitteltest2&quot;},&quot;targetId&quot;:&quot;mrec_4&quot;},{&quot;disablePsa&quot;:true,&quot;invCode&quot;:&quot;welt.de-amp-ros-mrec&quot;,&quot;sizes&quot;:[[300,250]],&quot;keywords&quot;:{&quot;kw_misc&quot;:[],&quot;pt0&quot;:96449,&quot;pt1&quot;:&quot;werbemitteltest2&quot;},&quot;targetId&quot;:&quot;mrec_5&quot;}]}">
  | </amp-ad>
```

**Comment by renebaudisch** [comment]
THX, I have to provide this to the publisher, but that makes sense...

**Comment by renebaudisch** [comment]
The publisher now made a new testpage https://amp.welt.de/amp/article/180940188.
The ast.js is throwing an error,
image

**Comment by matthewlane** [comment]
@renebaudisch We have https://github.com/ampproject/amphtml/pull/17261 waiting for review to hopefully fix this error

**Comment by liamqma** [comment]
Hi @renebaudisch and @matthewlane,

Same issue here.

```Javascript
[20:58:28:170] GENERAL_ERROR: events._dispatch: Error executing event handler function: global is not defined
```

I've setup a codepen example at https://codepen.io/liamqma/pen/yKrbPW.

**Pull request merged**